lib/Data/ObjectDriver/SQL.pm

```diff
@@ -247,25 +247,64 @@ sub _mk_term {
             $term = join " $logic ", @terms;
         } else {
             $col = $m->($col) if $m = $stmt->column_mutator;
-            $term = "$col IN (".join(',', ('?') x scalar @$val).')';
+            $term = $stmt->_mk_term_arrayref($col, 'IN', $val);
             @bind = @$val;
         }
     } elsif (ref($val) eq 'HASH') {
         my $c = $val->{column} || $col;
         $c = $m->($c) if $m = $stmt->column_mutator;
-        $term = "$c $val->{op} ?";
-        push @bind, $val->{value};
+        my $op = uc $val->{op};
+        if (($op eq 'IN' or $op eq 'NOT IN') and ref $val->{value} eq 'ARRAY') {
+            $term = $stmt->_mk_term_arrayref($c, $op, $val->{value});
+            push @bind, @{$val->{value}};
+        } elsif (($op eq 'IN' or $op eq 'NOT IN') and ref $val->{value} eq 'REF') {
+            my @values = @{${$val->{value}}};
+            $term = "$c $op (" . (shift @values) . ")";
+            push @bind, @values;
+        } elsif ($op eq 'BETWEEN' and ref $val->{value} eq 'ARRAY') {
+            Carp::croak "USAGE: foo => {op => 'BETWEEN', value => [\$a, \$b]}" if @{$val->{value}} != 2;
+            $term = "$c $op ? AND ?";
+            push @bind, @{$val->{value}};
+        } else {
+            if (ref $val->{value} eq 'SCALAR') {
+                $term = "$c $val->{op} " . ${$val->{value}};
+            } else {
+                $term = "$c $val->{op} ?";
+                push @bind, $val->{value};
+            }
+        }
     } elsif (ref($val) eq 'SCALAR') {
         $col = $m->($col) if $m = $stmt->column_mutator;
         $term = "$col $$val";
+    } elsif (ref($val) eq 'REF') {
```

Comment:
@charsbar 
I think we should also apply `column_mutator` at this block.
```perl
        $col = $m->($col) if $m = $stmt->column_mutator;
```

Reply:
Thanks!